source/dotnet/Tests/Application/BatchApplicationServiceTests.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using AutoFixture.Xunit2;
+using Energinet.DataHub.Core.TestCommon.AutoFixture.Attributes;
+using Energinet.DataHub.Wholesale.Application.Batches;
+using Energinet.DataHub.Wholesale.Domain.BatchAggregate;
+using Energinet.DataHub.Wholesale.Domain.GridAreaAggregate;
+using Energinet.DataHub.Wholesale.Domain.ProcessAggregate;
+using Energinet.DataHub.Wholesale.Tests.Domain.BatchAggregate;
+using FluentAssertions;
+using Moq;
+using NodaTime;
+using Xunit;
+using Xunit.Categories;
+
+namespace Energinet.DataHub.Wholesale.Tests.Application;
+
+[UnitTest]
+public class BatchApplicationServiceTests
+{
+    [Theory]
+    [InlineAutoMoqData]
+    public async Task SearchAsync_NoMatchingBatches_ReturnsZeroBatches(
+        [Frozen] Mock<IBatchRepository> batchRepositoryMock,
+        BatchApplicationService sut)
+    {
+        // Arrange
+        var noBatches = new List<Batch>();
+        batchRepositoryMock.Setup(x => x.GetAsync(It.IsAny<Instant>(), It.IsAny<Instant>())).ReturnsAsync(noBatches);
+
+        // Act
+        var batchSearchDto = new BatchSearchDto(DateTimeOffset.Now, DateTimeOffset.Now);
+        var searchResult = await sut.SearchAsync(batchSearchDto);
+
+        // Assert
+        searchResult.Count().Should().Be(0);
+    }
+
+    [Theory]
+    [InlineAutoMoqData]
+    public async Task SearchAsync_NoMatchingBatches_DoesThrowException(
+        [Frozen] Mock<IBatchRepository> batchRepositoryMock,
+        BatchApplicationService sut)
+    {
+        // Arrange
+        var noBatches = new List<Batch>();
+        batchRepositoryMock.Setup(x => x.GetAsync(It.IsAny<Instant>(), It.IsAny<Instant>())).ReturnsAsync(noBatches);
+
+        // Act
+        var batchSearchDto = new BatchSearchDto(DateTimeOffset.Now, DateTimeOffset.Now);
+        var exception = await Record.ExceptionAsync(() => sut.SearchAsync(batchSearchDto));
+
+        // Assert
+        Assert.Null(exception);
```

Comment:
Alternative approach (written from the top of my head):
```suggestion
        var batchSearchDto = new BatchSearchDto(DateTimeOffset.Now, DateTimeOffset.Now);
        
        // Act and assert
        Assert.DosNotThrowAsync(async () => sut.SearchAsync(batchSearchDto));
```

Reply:
Isn't that removed? See ttps://github.com/xunit/xunit/issues/188

Reply:
I removed assert and commented on the fact that the test will fail if it has an exception